cypress-E2E-tests/cypress/integration/asset_properties_widget.cy.ts

```diff
@@ -0,0 +1,614 @@
+    cy.get(assetsTextElement).should('contain.text', 'Assets');
+    cy.get(asset_properties_widget_elements.loadMoreButton).click();
+    cy.get(`p[title='${assetName}']`).click();
+    cy.get(textElement).should('contains.text', 'Assets > ' + assetName);
+    cy.get(backArrowElement).should('exist');
+    cy.get('input[placeholder="Filter this column…"]').click();
+    cy.get(filterElement).should('exist');
+    cy.get('.c8y-empty-state > .d-flex > .text-medium').should(
+      'have.text',
+      'No results to display.'
+    );
+    cy.get('p.small').should('have.text', 'The selected asset has no children.');
+  });
+
+  // Verify that unchecking the property reflects under Properties section
+  it('TC_Asset_Properties_Widget_config_031', () => {
+    const checkboxSelectedElement = ':nth-child(1) > [style="width: 10%;"] > .ng-untouched';
+    const checkboxUnSelectElement = ':nth-child(1) > [style="width: 10%;"] > .ng-valid';
+    cy.get(checkboxSelectedElement).click();
+    cy.get(checkboxUnSelectElement).should('exist');
+  });
+
+  // Verify that cancel button works as expected
+  it('TC_Asset_Properties_Widget_config_032', () => {
+    cy.get(asset_properties_widget_elements.cancelButton).click();
+    cy.get(asset_properties_widget_elements.addWidgetButton).should('exist');
+  });
+
+  // Verify the presence of cancel and save button.
+  it('TC_Asset_Properties_Widget_config_033', () => {
+    cy.get(asset_properties_widget_elements.cancelButton).should('exist');
+    cy.get(asset_properties_widget_elements.saveButton).should('exist');
+  });
+
+  // Asset property widget title can be duplicate
+  it('TC_Asset_Properties_Widget_config_034', () => {
+    cy.selectAssetAndSave(assetName);
+    cy.get(asset_properties_widget_elements.widgetDashboardAddWidgetButton).click();
+    cy.get(asset_properties_widget_elements.cardElement).click();
+    cy.selectAssetAndSave(assetName);
+    cy.deleteWidgetInstances([assetProperties, assetProperties]);
+  });
+
+  // Verify the presence "No widgets to display" meessage in the dashboard
+  it('TC_Asset_Properties_Widget_view_001', () => {
+    cy.get(asset_properties_widget_elements.cancelButton).click();
+    const message = 'No widgets to display';
+    cy.contains(message);
+  });
+
+  // Verify the presence of Name,ID and Type rows property fields.
+  it('TC_Asset_Properties_Widget_view_002', () => {
+    cy.selectAssetAndSave(assetName);
+    cy.get(cardTitleElement).should('contain.text', assetProperties);
+    cy.get("p[title='Name']").should('exist');
+    cy.get("p[title='ID']").should('exist');
+    cy.get("p[title='Asset model']").should('exist');
+    cy.deleteCard();
+  });
+
+  // Verify the presence of settings symbol at the top. User should be able to see edit and remove option on clicking it.
+  it('TC_Asset_Properties_Widget_view_003', () => {
+    cy.selectAssetAndSave(assetName);
+    cy.get(asset_properties_widget_elements.settingsButton).should('exist').click();
```

Comment:
Wherever you are expecting the element to be visible on the UI then add **.should('be.visible')**